tests/e2e/chain/config.go

```diff
@@ -0,0 +1,335 @@
+
+	// Currently only running one chain, so this is not used
+	// chainB
+	ChainBID      = "evmos_9000-1"
+	CoinBalanceB  = 5000000000000000000
+	StakeBalanceB = 4400000000000000000
+	StakeAmountB  = 4000000000000000000
+)
+
+var (
+	StakeAmountIntA  = sdk.NewInt(StakeAmountA)
+	StakeAmountCoinA = sdk.NewCoin(StakeDenom, StakeAmountIntA)
+	StakeAmountIntB  = sdk.NewInt(StakeAmountB)
+	StakeAmountCoinB = sdk.NewCoin(StakeDenom, StakeAmountIntB)
+
+	InitBalanceStrA = fmt.Sprintf("%d%s,%d%s", CoinBalanceA, CoinDenom, StakeBalanceA, StakeDenom)
+	InitBalanceStrB = fmt.Sprintf("%d%s,%d%s", CoinBalanceB, CoinDenom, StakeBalanceB, StakeDenom)
+)
+
+func addAccount(path, moniker, amountStr string, accAddr sdk.AccAddress) error {
+	serverCtx := server.NewDefaultContext()
+	config := serverCtx.Config
+
+	config.SetRoot(path)
+	config.Moniker = moniker
+
+	coins, err := sdk.ParseCoinsNormalized(amountStr)
+	if err != nil {
+		return fmt.Errorf("failed to parse coins: %w", err)
+	}
+
+	balances := banktypes.Balance{Address: accAddr.String(), Coins: coins.Sort()}
+	genAccount := authtypes.NewBaseAccount(accAddr, nil, 0, 0)
+
+	genFile := config.GenesisFile()
+	appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+	}
+
+	authGenState := authtypes.GetGenesisStateFromAppState(util.Cdc, appState)
+
+	accs, err := authtypes.UnpackAccounts(authGenState.Accounts)
+	if err != nil {
+		return fmt.Errorf("failed to get accounts from any: %w", err)
+	}
+
+	if accs.Contains(accAddr) {
+		return fmt.Errorf("failed to add account to genesis state; account already exists: %s", accAddr)
+	}
+
+	// Add the new account to the set of genesis accounts and sanitize the
+	// accounts afterwards.
+	accs = append(accs, genAccount)
+	accs = authtypes.SanitizeGenesisAccounts(accs)
+
+	genAccs, err := authtypes.PackAccounts(accs)
+	if err != nil {
+		return fmt.Errorf("failed to convert accounts into any's: %w", err)
+	}
+
+	authGenState.Accounts = genAccs
+
+	authGenStateBz, err := util.Cdc.MarshalJSON(&authGenState)
+	if err != nil {
+		return fmt.Errorf("failed to marshal auth genesis state: %w", err)
+	}
+
+	appState[authtypes.ModuleName] = authGenStateBz
+
+	bankGenState := banktypes.GetGenesisStateFromAppState(util.Cdc, appState)
+	bankGenState.Balances = append(bankGenState.Balances, balances)
+	bankGenState.Balances = banktypes.SanitizeGenesisBalances(bankGenState.Balances)
+
+	bankGenStateBz, err := util.Cdc.MarshalJSON(bankGenState)
+	if err != nil {
+		return fmt.Errorf("failed to marshal bank genesis state: %w", err)
+	}
+
+	appState[banktypes.ModuleName] = bankGenStateBz
+
+	appStateJSON, err := json.Marshal(appState)
+	if err != nil {
+		return fmt.Errorf("failed to marshal application genesis state: %w", err)
+	}
+
+	genDoc.AppState = appStateJSON
+	return genutil.ExportGenesisFile(genDoc, genFile)
+}
+
+func initGenesis(c *internalChain) error {
```

Comment:
we also need to update the denominations of all the corresponding modules (evm, gov, etc)